ichk/s3_resource_interface.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import hashlib
+import os
+
+import boto3
+import botocore.exceptions
+
+from irods.models import Resource
+import irods.exception as iexc
+
+from ichk.resource_interface import ResourceInterface
+from ichk.status_codes import Status
+
+class S3ResourceInterface(ResourceInterface):
+    CHUNK_SIZE = 8192
+
+
+    def __init__(self, irods_session, resource_name):
+        self.irods_session = irods_session
+        self.resource_name = resource_name
+        self.s3_hostname = self._get_s3_hostname(resource_name)
+        (self.s3_accesskey, self.s3_secretkey) = self._get_s3_credentials(resource_name)
+        self.s3_region = self._get_resource_context_param(resource_name, "S3_REGIONNAME")
+        boto3_session = boto3.Session(aws_access_key_id = self.s3_accesskey.strip(),
+                                   aws_secret_access_key = self.s3_secretkey.strip(),
+                                   region_name = self.s3_region.strip())
+        self.boto3_client = boto3_session.client('s3', endpoint_url="https://" + self.s3_hostname)
```

Comment:
Can we build the endpoint url using S3_PROTO? To support http S3 resources (for development & testing).

Reply:
Thanks, it uses S3_PROTO now.